apps/api/src/notifications/providers/notifications.sendgrid.provider.ts

```diff
@@ -180,4 +269,53 @@ export class SendGridNotificationsProvider
 
     return response
   }
+
+  async sendBulkEmail(data: MassMailDto, contactsMap: ContactsMap[], value: string): Promise<void> {
+    const currentDate = new Date()
+    contactsMap.forEach((contacts, index) => {
+      //Schedule  batches in a minute difference
+      currentDate.setMinutes(currentDate.getMinutes() + index)
```

Comment:
This will result in current date with the following values (say it is now 11:00):

11:00
11:01 
11:03
11:06
11:10
11:15
11:21
...

If I am reading this correctly.
It will probably work, I am just not sure this was the intent, especially if we expect this list to be big...

Reply:
Will have to revisit whether the scheduling part works as intended , but the idea is to schedule each call in a minute difference, due to Sendgrid Rate limiting. eg:
First batch of emails: 10:00
Second batch of emails: 10:01
Third batch of emails: 10:02

Big lists are not a problem since those emails are sent in bulk. What the implementation does is the following:
Get contacts data from list -> Split contact's data into chunks, with each chunk having up to 1000 emails -> Schedule mail send for the emails within each chunk.

Though thinking about it we should probably have a way to cancel the mail sent at some point.

Reply:
Lets test this in staging/prod environment? We already have a private contact list, to test this internally first?